domain-models-runtime/src/main/java/org/folio/rest/tools/utils/TenantLoading.java

```diff
@@ -373,11 +332,21 @@ private static void loadURL(Map<String, String> headers, URL url,
       log.warn("TenantLoading.perform No X-Okapi-Url header");
       return Future.failedFuture("No X-Okapi-Url header");
     }
-    Iterator<LoadingEntry> it = loadingEntries.iterator();
-    HttpClient httpClient = vertx.createHttpClient();
-    return performR(okapiUrl, ta, headers, it, httpClient, recordsLoaded)
-        .onComplete(complete -> httpClient.close());
-  }
+    WebClient httpClient = WebClient.create(vertx);
+    Future<Integer> future = Future.succeededFuture(recordsLoaded);
+    for (LoadingEntry entry : loadingEntries) {
+      if (ta != null) {
+        final String okapiUrlFinal = okapiUrl;
+        for (Parameter parameter : ta.getParameters()) {
+          if (entry.key.equals(parameter.getKey()) && "true".equals(parameter.getValue())) {
+            future = future.compose(x -> loadData(okapiUrlFinal, headers, entry, httpClient)
+                .compose(y -> Future.succeededFuture(x + y)));
```

Comment:
Rename x and y, and use map():
```suggestion
            future = future.compose(sum -> loadData(okapiUrlFinal, headers, entry, httpClient)
                .map(newRecords -> sum + newRecords));
```